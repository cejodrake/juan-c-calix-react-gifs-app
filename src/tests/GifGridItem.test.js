import React from 'react'
import {shallow } from  'enzyme'
import {GifGridItem} from '../components/GifGridItem'


describe("pruebas en <GifGridItem/>", () => {

const title ="santiago ";
const url = "https://localhost/2.jpg"
const wrapper = shallow(<GifGridItem title={title} url={url}/>)

  test('debe de mostrar el componente correctamente', () =>{

      expect(wrapper).toMatchSnapshot()

  })
  test('debe de tener un parrafo con el title',()=>{
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe("santiago");
  })
  test('debe de teber la imagen igual al url y alt de los pros ', ()=> {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);

  })

  test('debe de tener animate__fadeIn',()=>{
    const div = wrapper.find('div')
    const className = div.prop('className')
     expect(className.includes('animate__fadeIn')).toBe(true)
  })

})
