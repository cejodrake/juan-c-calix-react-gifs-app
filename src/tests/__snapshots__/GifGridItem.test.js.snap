// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGridItem/> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="santiago "
    src="https://localhost/2.jpg"
  />
  <p>
     
    santiago 
     
  </p>
</div>
`;
